src/KeyManager.php

```diff
@@ -63,8 +69,8 @@ public function getKeysByStorageMethod($storage_method) {
     });
 
     $keys = [];
-    foreach ($key_types as $key_type_id => $key_type) {
-      $keys = array_merge($keys, $this->getKeysByType($key_type_id));
```

Comment:
I discovered this bug while running tests. The output of getDefinitions() is in an indexed around and is not keyed by the plugin id. The plugin id from the annotation is inside the array as the 'id' key.